### PR TITLE
V6.1.0 rc

### LINKED ISSUE #3706: K version bump feature list (6.1.0)
Updated up to: 

- Features:
  - https://github.com/runtimeverification/k/pull/3487
  - https://github.com/runtimeverification/k/pull/3475
  - https://github.com/runtimeverification/k/pull/3506
  - https://github.com/runtimeverification/k/pull/3469
  - https://github.com/runtimeverification/k/pull/3562
  - https://github.com/runtimeverification/k/pull/3568
  - https://github.com/runtimeverification/k/pull/3575
  - https://github.com/runtimeverification/k/pull/3576
  - https://github.com/runtimeverification/k/pull/3581
  - https://github.com/runtimeverification/k/pull/3579
  - https://github.com/runtimeverification/k/pull/3589
  - https://github.com/runtimeverification/k/pull/3596
  - https://github.com/runtimeverification/k/pull/3603
  - https://github.com/runtimeverification/k/pull/3611
  - https://github.com/runtimeverification/k/issues/3619
  - https://github.com/runtimeverification/k/pull/3623
  - https://github.com/runtimeverification/k/pull/3638
  - https://github.com/runtimeverification/k/pull/3642
  - https://github.com/runtimeverification/k/pull/3639
  - https://github.com/runtimeverification/k/pull/3648
  - https://github.com/runtimeverification/k/pull/3678
  - https://github.com/runtimeverification/k/pull/3686
  - https://github.com/runtimeverification/k/pull/3687
  - https://github.com/runtimeverification/k/pull/3660
  - https://github.com/runtimeverification/k/pull/3600
  - https://github.com/runtimeverification/k/pull/3725
  - https://github.com/runtimeverification/k/pull/3736
  - https://github.com/runtimeverification/k/pull/3700
  - https://github.com/runtimeverification/k/pull/3761

- Bug fixes:
  - https://github.com/runtimeverification/k/pull/3484
  - https://github.com/runtimeverification/k/pull/3507
  - https://github.com/runtimeverification/k/pull/3561
  - https://github.com/runtimeverification/k/pull/3557
  - https://github.com/runtimeverification/k/pull/3063
  - https://github.com/runtimeverification/k/pull/3578
  - https://github.com/runtimeverification/k/pull/3584
  - https://github.com/runtimeverification/k/pull/3602
  - https://github.com/runtimeverification/k/pull/3618
  - https://github.com/runtimeverification/k/pull/3653
  - https://github.com/runtimeverification/k/pull/3654
  - https://github.com/runtimeverification/k/issues/3530
  - https://github.com/runtimeverification/k/pull/3701

To Check:

- [ ] File `./package/version` has been bumped to `NEW_MAJOR.NEW_MINOR.0`, and `./package/version.sh sub` has been invoked.
- [ ] `CHANGELOG.md` has been updated with new features.
- [ ] [K active projects](https://github.com/runtimeverification/k/blob/master/web/pages/projects.md) page is up-to-date (receiving commits in the last 3 months).
- [ ] Audit of warnings in regression test-suite and tutorial (inspect CI output for unintentional warnings).
- [ ] Audit package dependencies in `package/` (look at various package/dependency lists in `package/` subdirectory and think whether any of them is now not needed).

### PR DESCRIPTION
**Pull request by radumereuta**
Mostly incremental changes.

Closes: #3706

Merge only if the release works.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,66 @@
...
+Features
+--------
+- Added support for MacOS 13 Ventura and Debian 12 Bookworm. We dropped support for
+  Ubuntu 20 Focal. You can now build K from sources on Apple Silicon. See README
+  for more details.
+
+- Updated dependency to Java version 17 or higher.
+
+- Added the Haskell Backend Booster as a dependency to K. This can improve performance
+  when running large proofs. It combines the llvm-backend for concrete execution and
```
Comment (Baltoli) [inline CHANGELOG.md]:
```suggestion
  when running large proofs. It uses the LLVM Backend for concrete execution and
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,66 @@
...
+  relies on the haskell-backend to simplify terms when there is a split in the proof.
```
Comment (Baltoli) [inline CHANGELOG.md]:
```suggestion
  relies on the Haskell Backend to simplify terms when there is a split in the proof.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,66 @@
...
+
+- Added `--smt-timeout` flag to `krun` and `kprove`.
+
+- Changed the Maven repository to Cloudrepo for more stability and flexibility when
+  building K from sources.
+
+- Improved attribute error messages by creating a whitelist dependent on the context.
+  This check is now mandatory.
+
+- Rule label cannot contain ` or whitespace
```
Comment (Baltoli) [inline CHANGELOG.md]:
```suggestion
- Rule label can no longer contain backticks (`) or whitespace
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,66 @@
...
+- Added `--debugger-command` flag to krun.
+
+- Added two options `--debug-tokens` and `--debug-parse` to help with debugging
+  parsing errors. The first option will print a Markdown table with all the matched
+  tokens by the scanner. The second one will give more details about the partial parse
+  tree constructed before an error was encountered.
+
+Misc/Bug Fixes
+--------------
+- Fixed a bug where the llvm backend would segfault because of badly initialized
```
Comment (Baltoli) [inline CHANGELOG.md]:
```suggestion
- Fixed a bug where the LLVM backend would segfault because of badly initialized
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,66 @@
 copyright: Copyright (c) K Team. All Rights Reserved.
 ---
 
+K Framework 6.1.0
+=================
+
+Features
+--------
+- Added support for MacOS 13 Ventura and Debian 12 Bookworm. We dropped support for
```
Comment (Baltoli) [inline CHANGELOG.md]:
```suggestion
- Added support for MacOS 13 Ventura. We dropped support for
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,6 +2,66 @@
...
+  Ubuntu 20 Focal. You can now build K from sources on Apple Silicon. See README
```
Comment (Baltoli) [inline CHANGELOG.md]:
```suggestion
  Ubuntu 20.04 Focal. K can now be built from source on Apple Silicon. See README
```

**Review by Baltoli — APPROVED** [review]
A few small wording changes; otherwise looks good to me!

**Review by F-WRunTime — APPROVED** [review]
Looks good. Though were we discussing dropping Arch going forward? Has that decision been finalized? @Baltoli I think was discussing this last week.

**Review by Baltoli — APPROVED** [review]

**Pull request merged**